src/infinity-menu.js

```diff
@@ -261,6 +261,7 @@ export default class InfinityMenu extends React.Component {
 
 		/*header component*/
 		const headerProps = {
+			metadata: this.props.metadata,
```

Comment:
instead of another forced prop that the user must conform too, i think something like a prop called headerProps would be preferable. Then you can do something like this and the user can path through whatever arbitrary props they desire. Also the term metadata is very general, especially since it is only going to the header and is not general metadata for the whole component.
```js
		const headerProps = {
			metadata: this.props.metadata,
			isSearching: this.state.search.isSearching,
			searchInput: this.state.search.searchInput,
			setSearchInput: this.setSearchInput,
			stopSearching: this.stopSearching,
			startSearching: this.startSearching,
			onClose: this.props.onClose,
                       ...this.props.headerProps
		};
```

Whatever way this goes, this also needs documentation as this is an open source project...